src/Microsoft.Health.Fhir.Core/Features/Search/BundleFactory.cs

```diff
@@ -118,8 +118,16 @@ public Bundle CreateHistoryBundle(IEnumerable<Tuple<string, string>> unsupported
                 }
             }
 
+            IDictionary<string, object> routeValues = new Dictionary<string, object>();
+
+            // Add the _before value to the self link if we added it to the search parameters
+            if (addedBefore != null)
+            {
+                routeValues.Add(KnownQueryParameterNames.Before, addedBefore.Value.ToString("o"));
```

Comment:
The supplied _before parameter is a PartialDateTime but this is DateTimeOffset so the if the input was /Patient/_history?_before=2019 then the self-link returned in the bundle will be /Patient/_history?_before=2019-01-01T00:00:00.00000Z (or equivalent of that). I just want to make sure that's intentional.

Reply:
We should only add a new _before parameter if it was not specified by the user.  If the user specified a value for _before, it should remain on the selflink, unchanged

---
In reply to: [282223550](https://github.com/microsoft/fhir-server/pull/455#discussion_r282223550) [](ancestors = 282223550)

Reply:
I have reverted the code that added the _before param when none was specified.